core/tests/protractor_utils/general.js

```diff
@@ -154,7 +154,10 @@ var ensurePageHasNoTranslationIds = async function() {
   await waitFor.visibilityOf(
     oppiaBaseContainer,
     'Oppia base container taking too long to appear.');
-  var promiseValue = await oppiaBaseContainer.getAttribute('innerHTML');
+  let promiseValue = await browser.executeScript(() => {
+    return document.getElementsByClassName(
+      'protractor-test-base-container')[0].innerHTML;
+  });
```

Comment:
Why this change?

Reply:
`getAttribute('innerHTML')` was returning null.

Reply:
oh! do we know why? In general, we try to avoid "browser.executeScript" where possible because it could introduce flakiness. So this should be a last resort.

Reply:
The purpose of `getAttribute('innerHTML')` is to get contents of this attribute.
![image](https://user-images.githubusercontent.com/24855641/124656613-10636000-debf-11eb-9445-0f6d3787e750.png)

It is not similiar to javascript `innerHTML` property on html element.
![image](https://user-images.githubusercontent.com/24855641/124656763-3c7ee100-debf-11eb-8662-74125c220f1d.png)

This javascript property and html attribute are two different things for protractor.

It is returning null due to absence of any innerHTML html attribute on base-container div element.

Earlier, there used to be a `getInnerHTML` function provided by protractor.
This was removed and browser.executeScript was the replacement.

https://github.com/angular/protractor/blob/master/CHANGELOG.md#breaking-changes-3

![image](https://user-images.githubusercontent.com/24855641/124657216-d9da1500-debf-11eb-9cad-7f32180e43cd.png)

Reply:
Proof:
test html file.
![image](https://user-images.githubusercontent.com/24855641/124657676-72709500-dec0-11eb-8dac-620770e523da.png)

Test:
![image](https://user-images.githubusercontent.com/24855641/124657714-7ef4ed80-dec0-11eb-96a7-8d044c3941f0.png)

This proves that `innerHTML` attribute and innerHTML javascript property behaves differently from each other.

Reply:
Ah, I see. In that case, can you clearly document in a comment that the recommended protractor approach (as described above) is to use `browser.executeScript` when innerHTML needs to be read? Thanks!

Reply:
done